File: Ruby/civitas/lib/casilla_sorpresa.rb

#encoding: utf­8
require_relative "casilla.rb"
require_relative 'sorpresa_ir_a_carcel'
require_relative 'sorpresa_ir_a_casilla'
require_relative 'sorpresa_pagar_cobrar'
require_relative 'sorpresa_por_casa_hotel'
require_relative 'sorpresa_por_jugador'
require_relative 'sorpresa_salir_carcel'
require_relative 'sorpresa_calle'
require_relative 'sorpresa_convierte_calle'
module Civitas
  class CasillaSorpresa < Casilla
    
    def initialize(_mazo, _nombre)
      super(_nombre)
      @mazo = _mazo
      @sorpresa = Sorpresa.new("")
    end
  
  def recibe_jugador(iactual,todos)
    puts "En recibe jugador de sorpresa"
      if(jugador_correcto(iactual, todos))
        @sorpresa = @mazo.siguiente()
        informe(iactual, todos)
        puts @sorpresa.to_ss()
        @sorpresa.aplicar_a_jugador(iactual, todos)
      end
  end
    
  def to_sc()
    return "Sorpresa " + @nombre
  end
end
  end

File: Java/civitas/src/civitas/CasillaSorpresa.java
package civitas;

import java.util.ArrayList;

public class CasillaSorpresa extends Casilla{
    private MazoSorpresa mazo;
    private Sorpresa sorpresa;
    
    CasillaSorpresa(MazoSorpresa _mazoSorpresa, String _nombre){
        super(_nombre);
        mazo = _mazoSorpresa;
    }
    
    @Override
    public void recibeJugador(int iActual, ArrayList<Jugador> todos){
        if(super.jugadorCorrecto(iActual, todos)){
            sorpresa=mazo.siguiente();
            super.informe(iActual, todos);
            sorpresa.aplicarAJugador(iActual, todos);
        }
    }
    
    @Override
    public String toString(){
        return "Sorpresa " + sorpresa.getTexto();
    }
    
}
